File: src/me/krickl/memebotj/InternalCommands/ModeratorCommands/AutogreetCommand.scala
package me.krickl.memebotj.InternalCommands.ModeratorCommands

import me.krickl.memebotj.{Memebot, ChannelHandler, CommandHandler, UserHandler}


class AutogreetCommand(channel: String, command: String, dbprefix: String) extends CommandHandler(channel,
	command, dbprefix) {

	this.setAccess("moderators")

	this.setNeededCommandPower(0)

	this.setHelptext("Syntax: !autogreet <add/remove/get> <name>")

	override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
		try {
			val nameToModify = data(1).toLowerCase()
			var message = ""
			var user: UserHandler = null
			user = if (channelHandler.getUserList.containsKey(nameToModify)) channelHandler.getUserList.get(nameToModify) else new UserHandler(nameToModify,
				this.getChannelOrigin)
			if ((data(0) == "add" || data(0) == "edit") &&
					CommandHandler.checkPermission(sender.getUsername, 25, channelHandler.getUserList)) {
				var newValue = data(2)
				for (x <- 3 until data.length) {
					newValue = newValue + " " + data(x)
				}
				if (!user.newUser) {
					user.setAutogreet(newValue)
					user.writeDBUserData()
					message = Memebot.formatText(channelHandler.localisation.localisedStringFor("AUTOGREET_ADDED"), channelHandler, sender, this, false, Array())
				} else {
					message = Memebot.formatText(channelHandler.localisation.localisedStringFor("AUTOGREET_FAILED"), channelHandler, sender, this, false, Array())
				}
			} else if (data(0) == "remove" &&
					CommandHandler.checkPermission(sender.getUsername, 25, channelHandler.getUserList)) {
				if (!user.newUser) {
					user.setAutogreet("")
					user.writeDBUserData()
					message = Memebot.formatText(channelHandler.localisation.localisedStringFor("AUTOGREET_REMOVED"), channelHandler, sender, this, false, Array())
				}
			} else if (data(0) == "get") {
				if (!user.newUser) {
					channelHandler.sendMessage(Memebot.formatText(channelHandler.localisation.localisedStringFor("AUTOGREET_GET"), channelHandler, sender, this, false, Array(user.getUsername, f"${user.getAutogreet}")), this.getChannelOrigin)
				}
			} else if (data(0) == "toggle") {
				sender.setEnableAutogreets(!sender.getEnableAutogreets)
				channelHandler.sendMessage(Memebot.formatText(channelHandler.localisation.localisedStringFor("AUTOGREET_TOGGLE"), channelHandler, sender, this, false, Array(sender.screenName, f"${sender.enableAutogreets}")), this.getChannelOrigin)
			}
			channelHandler.sendMessage(message, this.getChannelOrigin)
		} catch {
			case e: ArrayIndexOutOfBoundsException => e.printStackTrace()
		}
	}
}

File: src/main/java/me/krickl/memebotj/Commands/Internal/AutogreetCommand.java
package me.krickl.memebotj.Commands.Internal;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.Commands.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.UserHandler;
import me.krickl.memebotj.Utility.CommandPower;

/**
 * This file is part of memebotj.
 * Created by unlink on 11/04/16.
 */
public class AutogreetCommand extends CommandHandler {

    public AutogreetCommand(ChannelHandler channelHandler, String commandName, String dbprefix) {
        super(channelHandler, commandName, dbprefix);
    }

    @Override
    public void overrideDB() {
        this.setHelptext(Memebot.formatText("AUTOGREET_SYNTAX", getChannelHandler(), null, this, true, new String[]{}, ""));
    }

    @Override
    public void commandScript(UserHandler sender, String[] data) {
        try {
            String nameToModify = data[1].toLowerCase();
            String message = "";
            UserHandler user = null;
            if (getChannelHandler().getUserList().containsKey(nameToModify)) {
                user = getChannelHandler().getUserList().get(nameToModify);
            } else {
                user = new UserHandler(nameToModify, getChannelHandler().getChannel());
            }
            if (((data[0].equals("add") || data[0].equals("edit")) && checkPermissions(sender, CommandPower.modAbsolute, CommandPower.modAbsolute))) {
                String newValue = data[2];
                for (int x = 3; x < data.length; x++) {
                    newValue = newValue + " " + data[x];
                }
                if (!user.isNewUser()) {
                    user.setAutogreet(newValue);
                    user.writeDB();
                    message = Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("AUTOGREET_ADDED"), getChannelHandler(), sender, this, false, new String[]{}, "");
                } else {
                    message = Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("AUTOGREET_FAILED"), getChannelHandler(), sender, this, false, new String[]{}, "");
                }
            } else if (data[0].equals("remove") && checkPermissions(sender, CommandPower.modAbsolute, CommandPower.modAbsolute)) {
                if (!user.isNewUser()) {
                    user.setAutogreet("");
                    user.writeDB();
                    message = Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("AUTOGREET_REMOVED"), getChannelHandler(), sender, this, false, new String[]{}, "");
                }
            } else if (data[0].equals("get")) {
                if (!user.isNewUser()) {
                    getChannelHandler().sendMessage(Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("AUTOGREET_GET"), getChannelHandler(), sender, this, false, new String[]{user.getUsername(), user.getAutogreet()}, getChannelHandler().getChannel()));
                }
            } else if (data[0].equals("toggle")) {
                sender.setEnableAutogreets(!sender.isEnableAutogreets());
                getChannelHandler().sendMessage(Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("AUTOGREET_TOGGLE"), getChannelHandler(), sender, this, false, new String[]{sender.getUsername(), Boolean.toString(sender.isEnableAutogreets())}, getChannelHandler().getChannel()));
            }
            getChannelHandler().sendMessage(message, getChannelHandler().getChannel());
        } catch(ArrayIndexOutOfBoundsException e) {
            e.printStackTrace();
        }
    }
}
